### DISCUSSION TITLE
Hangs Resuming from suspend

### DISCUSSION THREAD

**Discussion by nzrf** [i3/i3#5590] [Q&A] [answered]
I love i3; first off, the workflow is great for how I like to engage with my desktop. However, I've been facing an issue for some time where I am unable to put my computer to sleep, and when it resumes, everything hangs. I've tried different distros, from Fedora to Tumbleweed, and started i3 using various methods like gdm, lightdm, and just with xinitrc, even using default configs. Despite all these attempts, the problem persists, and the computer hangs intermittently after resuming.

I can induce the hanging issue after resume by following these steps:
1. Run "systemctl suspend" to put the computer to sleep.
2. Resume the computer with keyboard interaction.
3. To lock it up immediately, perform a rapid succession of i3 restarts.

I find it hard to believe that this problem is as widespread as it is popular; it must be something specific to my setup or hardware. My system specs are AMD 3950x with an Nvidia 3090 running driver version 535.86.05.

Can anyone help me troubleshoot and resolve this issue or seen this behavior?

**Comment by orestisfl** [comment]
Can you start by posting your xinitrc? When the computer is hanged, is it just i3 or everything? Can you switch to a new tty e.g. with ctrl+alt+F2?

**Comment by nzrf** [comment]
Created a new user with the default i3 configuration. After that I tried using both "exec i3" and "exec i3 --shmlog-size=26214400" in my .xinitrc file. However, both attempts have led to the same behavior resuming after suspend. I am still unable to switch to a new tty using keyboard input, although I can log in via SSH, and the computer remains responsive.

i3 version 4.22 and latest patches from tumbleweed. 

.xinitrc
`exec i3`
and 
.xinitrc
`exec i3 --shmlog-size=26214400
`
Both have resulted in the same behavior 

I was unable to switch to a new tty via the keyboard input after a suspend, but I am able to login via ssh and computer is responsive via ssh. 

I can send in whatever debug logs are needed.

**Reply by okraits** [reply]
Does it happen with a different window manager?

**Reply by orestisfl** [reply]
> I can send in whatever debug logs are needed.

Yes, please do. Just do everything you did above (with the `shmlog` flag) and once you ssh into your machine, run `DISPLAY=:0 i3-dump-log | bzip2 -c | curl --data-binary @- https://logs.i3wm.org`

See: https://i3wm.org/docs/debugging.html#_obtaining_the_debug_logfile

**Reply by orestisfl** [reply]
Additionally, the output of `ps aux | grep i3` could be useful

> Does it happen with a different window manager?

Finally, testing with a different wm (openbox, awesomewm are my usual choices) is a good idea.

**Reply by nzrf** [reply]
Tried awesome worked fine no hangs. Even on the same suspend/resume sequence which i3 had issues. Also tried subsequent suspends/resumes  on the  the same boot still no issues or even on fresh boot. 

I have also tried  X11 KDE and X11 gnome work fine as well.  
Even tried hyprland. I understand non x11, but worked fine suspending and resuming.

Here are the logs links below.
One interesting note is is there was LONG delay getting the logs when the display was hung and strange error message. 

before suspend https://logs.i3wm.org/logs/6245451133091840.bz2
after suspend log1 https://logs.i3wm.org/logs/6253272402755584.bz2

- Strange error message getting logs
```
i3-dump-log: Failed to write 1: Bad address
Data is not an i3 log file.
```

after suspend log2 https://logs.i3wm.org/logs/6225811589824512.bz2

**Comment by nzrf** [comment, marked as answer]
This can be marked resolved with hardware issues.
Combo of AMD 3950x w/ Gigabyte Motherboard + EVGA Nividia 3090 was causing the issue. Have since upgraded and no issues.
